src/rust/src/x509.rs

```diff
@@ -46,6 +83,115 @@ fn get_bit(input: &[u8], n: usize) -> bool {
     }
 }
 
+fn parse_name_attribute(
+    py: pyo3::Python<'_>,
+    attribute: AttributeTypeValue,
+) -> Result<pyo3::PyObject, PyAsn1Error> {
+    let x509_module = py.import("cryptography.x509")?;
+    let oid = x509_module
+        .call_method1("ObjectIdentifier", (attribute.type_id.to_string(),))?
+        .to_object(py);
+    let tag_enum = py
+        .import("cryptography.x509.name")?
+        .getattr("_ASN1_TYPE_TO_ENUM")?;
+    let py_tag = tag_enum.get_item(attribute.value.tag().to_object(py))?;
+    // TODO: is it safe to assume utf8? Current Python does.
```

Comment:
If teh current behaviors assumes this stuff is utf-8 it seems fine to me. We can always change this later.